runtime/rasdump/javadump.cpp

```diff
@@ -1468,19 +1468,102 @@ JavaCoreDumpWriter::writeEnvUserArgsHelper(J9VMInitArgs *vmArgs)
 	}
 
 	{
-		/* write ignored options */
-		bool anyIgnored = false;
+		/* Write ignored options.
+		 *
+		 * An option is printed as ignored if it has been consumed or if it is an -Xjit or
+		 * -Xaot option that fall under the rules below.
+		 *
+		 * Ignored options for -Xjit and -Xaot follows these rules:
+		 * 1. If -Xjit is followed by -Xint or -Xnojit,
+		 *    then any -Xjit:optionString beforehand appears as ignored.
+		 * 2. If -XX:+MergeCompilerOptions is present,
+		 *    then all -Xjit options after -Xint or -Xnojit should not be ignored.
+		 * 3. If multiple -Xjit strings appear, all but the last one will appear as ignored.
+		 *
+		 * Similar rules apply for -Xaot.
+		 */
+
+		jint numXjit = 0;
+		jint numXaot = 0;
+		bool hasXXMerge = false;
+
+		/* These indices ensure all the options before them will be printed as ignored. */
+		jint ignoreXjitBefore = 0;
+		jint ignoreXaotBefore = 0;
```

Comment:
Rather than counting `-Xjit` and `-Xaot` options, we could just keep track of the index of the last of each such option which should simplify the code to decide which options are ignored.